### PR TITLE
TST/BUG: remove override for `remote_file_list` tests, bugfix for single day requests

### PR DESCRIPTION
**Pull request by jklenzing**
Removes manual override of remote_file_list tests from the unit tests.  Requires pysat/pysat#652 to run correctly, since user and password are not passed through correctly.  Expected to fail on Travis CI until the above branch is merged, but can be used to track down remaining issues in #5.

Updates:
- addresses bug in single day requests for `remote_files_list`
- Migrates usage of deprecated `pysat._files` functions (moved to `pysat.utils.files`)

### CODE REVIEW

**Comment by aburrell** [comment]
@jklenzing I restarted the builds, now that the upstream changes have been made.  Was there anything else you wanted to add here?

File: pysatMadrigal/instruments/methods/madrigal.py [hunk 1]
```diff
@@ -491,14 +490,13 @@ def good_exp(exp, date_array=None):
...
-            exp_end = dt.datetime(exp.endyear, exp.endmonth, exp.endday,
-                                  exp.endhour, exp.endmin, exp.endsec)
+            exp_start = dt.date(exp.startyear, exp.startmonth,
+                                exp.startday)
+            exp_end = (dt.date(exp.endyear, exp.endmonth, exp.endday)
+                       + dt.timedelta(days=1))
 
             for date_val in date_array:
-                if date_val >= exp_start and date_val < exp_end:
+                if date_val.date() >= exp_start and date_val.date() <= exp_end:
```
Comment (jklenzing) [inline pysatMadrigal/instruments/methods/madrigal.py:499]: This was one of the sources of the bug.  If the experiment file starts partway through the first requested day, using datetime to compare will fail. ie, the code was being asked to evaluate `date_val >= exp_start` as
```
dt.datetime(2010, 4, 12, 0, 0, 0) >= dt.datetime(2010, 4, 12, 18, 34, 27)
```
Converting both to date and adding a pad for the end point to solve this issue.

File: pysatMadrigal/instruments/methods/madrigal.py [hunk 2]
```diff
@@ -436,14 +436,13 @@ def get_remote_filenames(inst_code=None, kindat=None, user=None,
                 date_array))
         start = date_array.min()
         stop = date_array.max()
-        if start == stop:
-            stop += dt.timedelta(days=1)
-
+    if start == stop:
+        stop += dt.timedelta(days=1)
```
Comment (jklenzing) [inline pysatMadrigal/instruments/methods/madrigal.py:440]: Regardless of whether date-array is specified, we need to check if a single day is requested and adjust accordingly.

**Comment by jklenzing** [comment]
@aburrell, I'm done changing things here.  There was a hiccup on the first round of tests that seems to be a server issue.  Restarting the job that failed.  Note that the travis tests have increased by 10-15 min.

**Review by aburrell — APPROVED** [review]
Thanks @jklenzing this is great.

**Pull request merged**